esp-wifi/src/compat/misc.rs

```diff
@@ -158,11 +158,41 @@ unsafe extern "C" fn strdup(str: *const i8) -> *const u8 {
 unsafe extern "C" fn atoi(str: *const i8) -> i32 {
     trace!("atoi {:?}", str);
 
-    unsafe {
-        let s = core::ffi::CStr::from_ptr(str);
-        let r = s.to_str().unwrap().parse().unwrap();
-        r
+    let mut sign: i32 = 1;
+    let mut res: i32 = 0;
+    let mut idx = 0;
+
+    // skip leading spaces
+    while str.add(idx).read_volatile() as u8 == b' ' {
```

Comment:
```suggestion
    while str.add(idx).read() as u8 == b' ' {
```

These don't have to be volatile I think

Reply:
Interesting: I assumed the code generated for `read` and `read_volatile` would be the same since the compiler can't make any assumption about the data behind the raw pointer but it's not 🤔 

Actually, the code for volatile is a bit smaller (0xca vs 0xb8 bytes for RV32-IMC using rustc 1.84.0-nightly (662180b34 2024-10-20)) - but probably not representative - just interesting

Reply:
Volatile is probably smaller because I think non-volatile may be optimized into 32-bit operations + handling remainder. But it's hard to know without looking at the assembly.